pkg/cri/resource-manager/requests.go

```diff
@@ -512,6 +513,61 @@ func (m *resmgr) RemoveContainer(ctx context.Context, method string, request int
 	return reply, rqerr
 }
 
+// ListContainers intercepts CRI requests for listing Containers.
+func (m *resmgr) ListContainers(ctx context.Context, method string, request interface{},
+	handler server.Handler) (interface{}, error) {
+
+	reply, rqerr := handler(ctx, request)
+
+	if rqerr != nil {
+		return reply, rqerr
+	}
+
+	if f := request.(*criapi.ListContainersRequest).Filter; f != nil {
+		switch {
+		case f.Id != "":
+			return reply, nil
+		case f.State != nil:
+			return reply, nil
+		case f.PodSandboxId != "":
+			return reply, nil
+		case len(f.LabelSelector) > 0:
+			return reply, nil
+		}
```

Comment:
I guess here one could use fallthrough
```
		switch {
		case f.Id != "",
		case f.State != nil,
		case f.PodSandboxId != "",
		case len(f.LabelSelector) > 0:
			return reply, nil
		}
```

but I don't know how much difference it really makes

Reply:
Hmm... I was under the impression that a case list is just syntactic sugar and it compiles to a similar/exactly same if-else if-else if construct than a normal switch without case lists.
 
And I guess this is waaay off-topic, but is that really called a `fallthrough` ? I was under the impression that a fallthrough refers to the explicit usage of the `fallthrough` keyword in golang, since unlike in C, by default the cases in a switch construct break instead of falling through.

But your suggestion has the benefit of emphasizing the identical return values in all those cases, so it's probably better in this case.

Reply:
> Hmm... I was under the impression that a case list is just syntactic sugar and it compiles to a similar/exactly same if-else if-else if construct than a normal switch without case lists.

Yes, I think this is the way it is. The idea behind my suggestion was just eliminating identical returns
 
> And I guess this is waaay off-topic, but is that really called a `fallthrough` ? I was under the impression that a fallthrough refers to the explicit usage of the `fallthrough` keyword in golang, since unlike in C, by default the cases in a switch construct break instead of falling through.

Yup, you're correct: fall-through only is the explicit use of `fallthrough` in one case, not combining multiple ifs in one case as a list. My terminology was bad, thanks for correcting me. 

> But your suggestion has the benefit of emphasizing the identical return values in all those cases, so it's probably better in this case.

This was the purpose of my nitty-gritty-suggestion, improving readability 😺  I wasn't sure if was good or bad, though.

Reply:
Okay, so that is syntactically incorrect go. What you can do is this:
```
	switch {
	case f.Id != "",
		f.State != nil,
		f.PodSandboxId != "",
		len(f.LabelSelector) > 0:
		    return reply, nil
	}
```
but that gets turned by the tabulation and formatting nazis into this:
```
	switch {
	case f.Id != "",
		f.State != nil,
		f.PodSandboxId != "",
		len(f.LabelSelector) > 0:
		return reply, nil
	}
```
which is not very readable. So I gave in a replaced it with an `if`.

Reply:
Woops, yes, my code example was borked 😬  And, nice work gofmt 😢  Maybe an if is actually the best here, agree